### PR TITLE
RAC-105: P & PM API endpoints returns quantified associations of parents

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

- Update Queries to return parent associations 
- Adapt tests

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | Ok
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php [hunk 1]
```diff
@@ -42,10 +44,11 @@ public function setUp(): void
     /**
      * @test
      */
-    public function testItReturnQuantifiedAssociationWithProductsOnSingleProduct()
+    public function itReturnQuantifiedAssociationWithProductsOnSingleProductWithParentQuantAssociationsOnProducts()
```
Comment (StevenVAIDIE) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:47]: Typo ?

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php [hunk 2]
```diff
@@ -112,6 +113,30 @@ public function testItReturnQuantifiedAssociationWithProductsOnMultipleProducts(
...
+                    ],
+                ]
+            )
+        );
+        $rootProductModel->setQuantifiedAssociations(
+            QuantifiedAssociations::createFromNormalized(
+                [
+                    'PRODUCT_SET' => [
+                        'products' => [
+                            ['identifier' => 'productB', 'quantity' => 7],
```
Comment (StevenVAIDIE) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:132]: Maybe add in rootProductModel :
                             ['identifier' => 'productC', 'quantity' => 5], 

And add in subProductModel :
                         ['identifier' => 'productA', 'quantity' => 3], 

Without change result expected

**Comment by StevenVAIDIE** [comment]
> You should also change src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Product/QuantifiedAssociation/GetProductModelQuantifiedAssociationsByProductIdentifiers.php
> 
> And for product model (same query than product but wiht only one left join with product model) :
> src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/ProductModel/QuantifiedAssociation/GetProductModelQuantifiedAssociationsByProductModelCodes.php
> src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/ProductModel/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductModelCodes.php

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php (hunk 1 above)
Comment (SamirBoulil) [reply tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:47]: yes

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php (hunk 2 above)
Comment (SamirBoulil) [reply tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:132]: heu... I don't understand :)

Comment (StevenVAIDIE) [reply tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:132]: To test desynchronisation between product and product models

Comment (SamirBoulil) [reply tests/back/Pim/Enrichment/Integration/Storage/Sql/Product/QuantifiedAssociation/GetProductQuantifiedAssociationsByProductIdentifiersIntegration.php:132]: I've added another test to check this case

**Pull request merged**